api/src/services/generate-xform.js

```diff
@@ -45,25 +45,30 @@ const transform = (formXml, stylesheet) => {
   });
 };
 
+const replaceLast = (haystack, needle) => {
+  const index = haystack.lastIndexOf(needle);
+  return haystack.slice(0, index) + haystack.slice(index + needle.length);
```

Comment:
If the need is not present in the function, this will return two copies of the string

```
replaceLast('abcaa ', 'z')
"abcaaabcaa "
```

Reply:
Fixed.